docs/DevGuide/PythonBindings.md

```diff
@@ -0,0 +1,131 @@
+
+Once a python module has been added you can write the actual bindings. You
+should structure your bindings directory to reflect the structure of the library
+your writing bindings for. For example, the bindings for `DataStructures` a
+source file exists for `DataVector` and any other classes we want bindings
+for such as `Matrix`. The functions that generate the bindings should be in the
+`py_bindings` namespace and have a reasonable name such as
+`bind_datavector`. There should be a file named `Bindings.cpp` which calls all
+the `bind_*` functions. The `Bindings.cpp` file is quite simple and should
+`include <boost/python.hpp>`, forward declare the `bind_*` functions, and then
+have `BOOST_PYTHON_MODULE` function. For example,
+
+\code{.cpp}
+#include <boost/python.hpp>
+
+namespace py_bindings {
+void bind_datavector();
+}  // namespace py_bindings
+
+BOOST_PYTHON_MODULE(_DataStructures) {
+  Py_Initialize();
+  py_bindings::bind_datavector();
+}
+\endcode
+
+Note that the library name is passed to `BOOST_PYTHON_MODULE` and is prefixed
+with an underscore. The underscore is important!
+
+The `DataVector` bindings serve as an example with code comments on how to write
+bindings for a class. There is also extensive documentation available directly
+from [Boost.Python]
+(https://www.boost.org/doc/libs/1_68_0/libs/python/doc/html/index.html)
+and [this GitHub repository](https://github.com/TNG/boost-python-examples) also
+has many helpful examples. SpECTRE currently aims support both Python 2.7 and
+Python 3 and as such all bindings must support both.
+
+\note Exceptions should be allowed to propagate through the bindings so that
+error handling via exceptions is possible from python rather than having the
+python interpreter being killed with a call to `abort`.
+
+## Testing Python Bindings and Code
+
+All the python bindings must be tested. SpECTRE uses the [unittest]() framework
+provided as part of python. To register a test file with CMake use the
+SpECTRE-provided function `spectre_add_python_test` passing as the first
+argument the test name (e.g. `"Unit.DataStructures.Python.DataVector"`), the
+file as the second argument (e.g. `Test_DataVector.py`), and a semicolon
+separated list of labels as the last (e.g. `"unit;datastructures;python"`).
+Unlike C++ tests, all the test cases should be in a single testing file and the
+python unit testing framework will run each test function individually. This
+helps reduce the time it takes to run tests since there is a non-negligible
+startup cost associated with launching a test.
+
+Below is an example of registering a python test file
+
+\code
+spectre_add_python_test(
+  "Unit.DataStructures.Python.DataVector"
+  Test_DataVector.py
+  "unit;datastructures;python")
+\endcode
+
+## Using The Bindings
+
+See \ref spectre_using_python "Using SpECTRE's Python"
+
+## Notes:
+
+- All python libraries are dynamic/shared libraries.
+- Exceptions should be allowed to propagate through the bindings so that
```

Comment:
This note is already above

Reply:
Yes, but I want it listed here explicitly so when someone just glances at this page and only looks at the notes they'll be aware of it :)